src/logger/src/lib.rs

```diff
@@ -133,663 +17,33 @@ extern crate serde_json;
-                ),
-                Box::new(
-                    OpenOptions::new()
-                        .read(true)
-                        .write(true)
-                        .open(metrics_file_temp.as_path())
-                        .unwrap()
-                ),
-            )
-            .is_ok());
-
-        info!("info");
-        warn!("warning");
-
-        let log_file_temp2 = TempFile::new().unwrap();
-        let metrics_file_temp2 = TempFile::new().unwrap();
-
-        assert!(l
-            .init(
-                &app_info,
-                Box::new(
-                    OpenOptions::new()
-                        .read(true)
-                        .write(true)
-                        .open(log_file_temp2.as_path())
-                        .unwrap()
-                ),
-                Box::new(
-                    OpenOptions::new()
-                        .read(true)
-                        .write(true)
-                        .open(metrics_file_temp2.as_path())
-                        .unwrap()
-                ),
-            )
-            .is_err());
-
-        info!("info");
-        warn!("warning");
-        error!("error");
-
-        // Here we also test that the last initialization had no effect given that the
-        // logging system can only be initialized with byte-oriented sinks once per program.
-        validate_logs(
-            &log_file,
-            &[
-                (TEST_INSTANCE_ID, "INFO", "lib.rs", "info"),
-                (TEST_INSTANCE_ID, "WARN", "lib.rs", "warn"),
-                (TEST_INSTANCE_ID, "INFO", "lib.rs", "info"),
-                (TEST_INSTANCE_ID, "WARN", "lib.rs", "warn"),
-                (TEST_INSTANCE_ID, "ERROR", "lib.rs", "error"),
-            ],
-        );
-
-        assert!(l.log_metrics().is_ok());
-
-        STATE.store(UNINITIALIZED, Ordering::SeqCst);
-
-        l.set_include_level(true);
-        l.set_include_origin(false, false);
-        let error_metadata = MetadataBuilder::new().level(Level::Error).build();
-        let log_record = log::Record::builder().metadata(error_metadata).build();
-        Logger::log(&l, &log_record);
-
-        assert_eq!(l.show_level(), true);
-        assert_eq!(l.show_file_path(), false);
-        assert_eq!(l.show_line_numbers(), false);
-
-        l.set_include_level(false);
-        l.set_include_origin(true, true);
-        let error_metadata = MetadataBuilder::new().level(Level::Info).build();
-        let log_record = log::Record::builder().metadata(error_metadata).build();
-        Logger::log(&l, &log_record);
-
-        assert_eq!(l.show_level(), false);
-        assert_eq!(l.show_file_path(), true);
-        assert_eq!(l.show_line_numbers(), true);
+use std::io::Write;
+use std::sync::{Mutex, MutexGuard};
+
+// Auxiliary function to flush a message to an entity implementing `Write` and `Send` traits.
+// This is used to either flush human-readable logs or metrics.
+fn write_to_destination(
```

Comment:
nit: I don't feel the need to encapsulate these two operations (`write` and `flush`) in a function. I think we can use them separated.